src/main/java/Duke.java

```diff
@@ -1,10 +1,130 @@
+import java.io.*;
+import java.util.Scanner;
+import java.util.*;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    public static void main(String[] args) throws IOException {
+        String line;
+        ArrayList<Task> tasks = new ArrayList<>();
+        Scanner in = new Scanner(System.in);
+        FileInputStream fis=new FileInputStream("ip\\src\\data\\duke.txt");
+        Scanner sc=new Scanner(fis);
+        FileWriter writer = new FileWriter("ip\\src\\data\\duke.txt");
+        final String SPLIT_LINE = "____________________________________________________________";
+        System.out.println(SPLIT_LINE);
+        System.out.println(" Hello! I'm Duke");
+        System.out.println(" What can I do for you?");
+        System.out.println(SPLIT_LINE);
+        if(sc.hasNextLine()){
+            line=sc.nextLine();}
+        else {
+            line=in.nextLine();
+        }
+        while (!line.equals("bye")){
+            try{
+                if(line.equals("list")){
+                    System.out.println(SPLIT_LINE);
+                    for (int i = 0; i < tasks.size(); i++) {
+
+                        System.out.println((i+1)+":"+tasks.get(i).toString());
+                    }
+                    System.out.println(SPLIT_LINE);
+                }else if(line.startsWith("done")){
+                    System.out.println(SPLIT_LINE);
+                    int taskNumberCompleted = Integer.parseInt(line.replaceAll("\\D+",""))-1;
+                    tasks.set(taskNumberCompleted,tasks.get(taskNumberCompleted).completeTask());
+                    System.out.println(" Nice! I've marked this task as done:");
+                    System.out.println(" "+":"+tasks.get(taskNumberCompleted));
+                    System.out.println(SPLIT_LINE);
+                } else if(line.startsWith("todo")){
+                    if(line.equals("todo")){
+                        throw new Exception();
+                    }else{
+                        System.out.println(SPLIT_LINE);
+                        System.out.println(" Got it. I've added this task:");
+                        tasks.add(new Todo(line));
+                        System.out.println("  "+new Todo(line));
+                        System.out.println(" Now you have " + tasks.size() + " tasks in the list.");
+                        System.out.println(SPLIT_LINE);
+                    }
+//                System.out.println(SPLIT_LINE);
+//                System.out.println("added: "+line);
+//                tasks.add(new Task(line));
+//                System.out.println(SPLIT_LINE);
```

Comment:
This seems to be past code that is commented out as it is no longer needed. You might want to delete this, and make full use of version control to keep track of dead code.